hvpy/api_groups/jpeg2000/tests/test_get_jp2_image.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+
+import pytest
+
+from hvpy.api_groups.jpeg2000.get_jp2_image import getJP2ImageInputParameters
+from hvpy.core import execute_api_call
+
+
+# Test Response
+def test_str_response():
+    date_obj = datetime(2022, 1, 1, 23, 59, 59)
+    params = {"date": date_obj, "sourceId": 14, "jpip": True, "Json": False}
+    params = getJP2ImageInputParameters(**params)
+    response = execute_api_call(input_parameters=params)
+    assert isinstance(response, str)
+    assert response.startswith("jpip://")
+
+
+def test_json_response():
+    date_obj = datetime(2022, 1, 1, 23, 59, 59)
+    params = {"date": date_obj, "sourceId": 14, "jpip": True, "Json": True}
+    params = getJP2ImageInputParameters(**params)
+    response = execute_api_call(input_parameters=params)
+    assert isinstance(response, dict)
+    assert "uri" in response
+    assert response["uri"].startswith("jpip://")
+
+
+def test_raw_response():
+    date_obj = datetime(2022, 1, 1, 23, 59, 59)
+    params = {"date": date_obj, "sourceId": 14, "jpip": False, "Json": False}
+    params = getJP2ImageInputParameters(**params)
+    response = execute_api_call(input_parameters=params)
+    assert isinstance(response, bytes)
+
+
+def test_raw_response_with_json():
+    date_obj = datetime(2022, 1, 1, 23, 59, 59)
+    params = {"date": date_obj, "sourceId": 14, "jpip": False, "Json": True}
+    params = getJP2ImageInputParameters(**params)
+    response = execute_api_call(input_parameters=params)
+    assert isinstance(response, bytes)
+
+
+def test_default_response():
+    date_obj = datetime(2022, 1, 1, 23, 59, 59)
+    params = {"date": date_obj, "sourceId": 14}
+    params = getJP2ImageInputParameters(**params)
+    response = execute_api_call(input_parameters=params)
+    assert isinstance(response, bytes)
+
+
+# Test error
+def test_error_handling():
+    params = {"sourceId": 14, "jpip": True, "Json": True}
+    with pytest.raises(ValueError):
```

Comment:
Can you match the message it raises too?

Reply:
sure

Reply:
I tried to match by storing an error message in a variable and did: 
``` with pytest.raises(ValidationError, match=error_message):```
But I got this error:
```shell
AssertionError: Regex pattern '1 validation error for getJP2ImageInputParameters\ndate\n  field required (type=value_error.missing)' does not match '1 validation error for getJP2ImageInputParameters\ndate\n  field required (type=value_error.missing)'. Did you mean to `re.escape()` the regex?
```

Reply:
You don't need to match the entire message, just the part that is important. 

What is the value of error_message in `match=error_message` here?

Reply:
I tried with:
```py
error_message = "1 validation error for getJP2ImageInputParameters\ndate\n  field required (type=value_error.missing)"
```
but got the error, so I tried 
```py
error_message = "1 validation error for getJP2ImageInputParameters\ndate\n  field required"
```
And that worked, so is this the correct way to do it?

Reply:
Just match the part of the message you want to check. It doesn't need to be the full message.